network-ntt/src/server.rs

```diff
@@ -0,0 +1,288 @@
+                                        lwcid,
+                                        Response::Ok(BlockHeaders(hdrs)),
+                                    );
+                                    sink1.unbounded_send(msg).map_err(|err| err.to_string())
+                                })
+                        })
+                        .map_err(move |msg| {
+                            let msg = Message::BlockHeaders(lwcid, Response::Err(msg));
+                            sink2.unbounded_send(msg).unwrap_or_else(|_| ());
+                        })
+                        .then(move |_| {
+                            sink3
+                                .unbounded_send(Message::CloseConnection(lwcid))
+                                .unwrap_or(());
+                            future::ok(())
+                        })
+                    }))
+                }
+                Inbound::GetBlocks(lwcid, get_blocks) => {
+                    let sink = sink_tx.clone();
+                    future::Either::B(future::Either::B(future::Either::A(
+                        server
+                            .node
+                            .block_service()
+                            .expect("block service is not implemented")
+                            .stream_blocks_to(&vec![get_blocks.from], &get_blocks.to)
+                            .map_err(|_| ())
+                            .and_then(move |blocks| {
+                                let inner1 = sink.clone();
+                                let inner2 = sink.clone();
+                                blocks
+                                    .map_err(|_| ())
+                                    .for_each(move |blk| {
+                                        inner1
+                                            .unbounded_send(Message::Block(
+                                                lwcid,
+                                                Response::Ok(blk),
+                                            ))
+                                            .map_err(|_| ())
+                                    })
+                                    .then(move |_| {
+                                        inner2.unbounded_send(Message::CloseConnection(lwcid))
+                                    })
+                                    .or_else(|_| Ok(()))
+                            })
+                            .or_else(|_| Ok(())),
+                    )))
+                }
+                Inbound::SendTransaction(_lwcid, tx) => {
+                    future::Either::B(future::Either::B(future::Either::B(
+                        server
+                            .node
+                            .transaction_service()
+                            .unwrap()
+                            .propose_transactions(&vec![tx])
+                            .and_then(|_| Ok(()))
+                            .or_else(|_| Ok(())),
```

Comment:
Um... I think there should be a FIXME comment to add proper error handling later, at least.